### DISCUSSION TITLE
TRRSケーブルで繋いでいる側のキーボードが時々反応しなくなる

### DISCUSSION THREAD

**Discussion by tongari07** [Yowkees/keyball#578] [Q&A]
## 動作環境

Keyball44
MacOS 14.4.1 (23E224)
[こちらのPRで実装されているファームウェアを書き込んで利用](https://github.com/tongari07/keyball/pull/1)
[Pro Micro Type C版](https://shop.yushakobo.jp/products/3905)

## 事象

左側にTypeCケーブルを繋ぎ、TRRSケーブルで右側を繋いでいます。
利用しているとたまに右側キーボードが効かなくなり、数秒間トラックボールのキーボードも効かなくなります。
LEDとOLEDは点灯しているので通電はしていると思われますが、LEDは正常に動いている時と違う配色になります。
数秒経てば治るのですが、結構頻繁に（体感10分おきくらい）に発生するので困っています。
何か情報があればご教授いただけると助かります

## 追記

[こちらの記事](https://zenn.dev/koron/articles/9fee38469a8acc#type-c-%2B-3.3v%E9%A7%86%E5%8B%95%E3%81%AE%E4%BA%92%E6%8F%9B%E6%A9%9F%E3%81%AE%E5%AD%98%E5%9C%A8)を参考に画像のように左右のPro Microのジャンパをハンダでショートさせ、5Vを有効にしたところとりあえず30分ほど発生しなくなりました。

![IMG_3452 copy](https://github.com/Yowkees/keyball/assets/52874708/1023777a-d717-41fc-b880-8195d30cc424)

ただし最初にTRRSケーブルを接続したときにMacから`USB Accessories Disabled` というエラーが出てしばらく接続されない事象がありました。数秒したら接続されるのですが、少々怖いので、もしどなたかご存知の方がいらっしゃったら何故発生するのかご教授いただけると助かります。

しばらくこの状態で運用してみて問題がなさそうであれば等DiscussionはCloseさせていただきたく思います。

**Comment by apio1969** [comment]
こんにちは。
全く同じ症状ではありませんが、私も似たような症状で悩んだことがあります。
私の場合は、TRRSケーブルで繋がった方が全く動かなくなりました。TRRSケーブルを何度か抜き差しすることで解決する事がありましたが、不定期にこの症状が現れるので、何度かダイオードやTRRSコネクタのハンダ部分を再加熱したりしてました。
それでも解決しなかったのですが、TRRSケーブルを何本か試してみたところ、この問題が解消しました。
結果的に種類の違うTRRSケーブル5本くらい購入して試したところ、そのうちの2本は症状が出ないケーブルで、現在はそのケーブルを使っています。（Amazonでいろいろ買いました。）
改善するかどうかはわかりませんが、可能性の一つとして試してみてはいかがでしょうか？

**Reply by tongari07** [reply]
ご共有ありがとうございます！
現在使っているケーブルが自作したTRRSケーブルなのもあり、正直それが原因かなとも思っていました。
ただ手持ちの別の既製品のものを使っても発生するので、いまいち原因を特定しきれていない状況です。
apio1969さんも5本中3本はダメだったということなので、私の方でもいくつか試してみようと思います。

**Comment by misonodenpati** [comment]
まさに同じような症状で、TRRSケーブルで繋いでいる側の動作が不安定でした。テスタでVCC電圧を当たってみたところトラックボールを弄ったりLEDをチカチカさせるたびにVCC電圧が振れていて３.１Vに振れた辺りでPROMICROがフリーズするような動作をしていました。PROMICROにはんだでブリッジを追加し、VCC電圧が4.6V程度に変わったことを確認後、取り付けると動作が安定しました。LEDを実装しなかった個体では問題が出ていなかったので負荷側の状態に依存しているかもしれません。ありがとうございました。
